OgreMain/include/OgreGpuProgram.h

```diff
@@ -478,13 +478,9 @@ namespace Ogre
             return mVpAndRtArrayIndexFromAnyShader;
         }
 
-        /** Sets whether this geometry program requires adjacency information
-            from the input primitives.
-        */
+        /// @deprecated
```

Comment:
Regarding deprecation we introduced the `OGRE_DEPRECATED_VER( n )` macro where n is the major version in which it was deprecated.

Thus the code should look like this:

```cpp
OGRE_DEPRECATED_VER( 4 ) virtual bool isAdjacencyInfoRequired()
```

(master branch is currently 4)